app/scripts/metamask-controller.js

```diff
@@ -319,9 +323,8 @@ export default class MetamaskController extends EventEmitter {
       {
         onPreferencesStateChange: (listener) =>
           this.preferencesController.store.subscribe(listener),
-        onNetworkStateChange: (cb) =>
-          this.networkController.on(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
-            const networkState = this.networkController.store.getState();
+        onNetworkStateChange: (cb) => {
+          this.networkController.store.subscribe((networkState) => {
```

Comment:
Updated to listen to the network controller rather than `networkDidChange` because we're grabbing the network state and calling the callback with it anyway.